api/data_pipeline.py

```
@@ -407,9 +536,231 @@ def transform_documents_and_save_to_db(
+            transformed_docs = None
+            
+            # Ensure environment variables are loaded before transformation
+            from dotenv import load_dotenv
+            load_dotenv()
+            
+            # Check if environment variables are available during transform
+            openai_key_present = bool(os.environ.get('OPENAI_API_KEY'))
+            logger.info(f"OPENAI_API_KEY available during transform: {openai_key_present}")
+            
+            if not openai_key_present:
+                raise ValueError("OPENAI_API_KEY environment variable not found. Cannot proceed with embedding generation.")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)